src/index.ts

```diff
@@ -1,11 +1,8 @@
-import * as Types from "./types";
-import * as decorators from "./decorators";
-import * as factories from "./factories";
-import * as providers from "./providers";
+import {DependencyContainer} from "./types";
 import {instance} from "./dependency-container";
 
 export {DependencyContainer} from "./types";
-export {factories};
-export {providers};
-export {decorators};
```

Comment:
The reason this was done is to enforce separate `namespaces` for the different types rather than having them all together. Let me think on this for a little.

I assume your preference is to have them all in the root export object?

Reply:
I prefer it because it improves tree shaking. When I use rollup (or webpack) to bundle it only includes the decorators I'm using. I'm trying to reduce the final bundle size.

Reply:
That's a fair enough reason =]